src/formatDate.js

```diff
@@ -50,7 +50,54 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldSeparator = fromFormat[3];
+  const newSeparator = toFormat[3];
+
+  const fromArr = date.split(oldSeparator);
+  const lastIndex = toFormat.length - 1;
+
+  const oldDay = fromArr[fromFormat.findIndex((item) => item.includes('D'))];
+  const oldMonth = fromArr[fromFormat.findIndex((item) => item.includes('M'))];
+  const oldYear = fromArr[fromFormat.findIndex((item) => item.includes('Y'))];
+
+  const result = [];
+
+  for (let i = 0; i < lastIndex; i++) {
+    let current = '';
+
+    switch (toFormat[i]) {
+      case 'DD':
+        current = oldDay;
+        break;
+
+      case 'MM':
+        current = oldMonth;
+        break;
+
+      case 'YY':
+        current = oldYear.length > 2
+          ? current = oldYear.slice(2)
+          : current = oldYear;
+        break;
+
+      case 'YYYY':
+        if (oldYear.length < 4) {
+          current = Number(oldYear >= 30)
+            ? current = String(19 + oldYear)
+            : current = String(20 + oldYear);
```

Comment:
Why do you repeat code? Use ternary only for finding 19 || 20.
Why do you use assignments inside the ternary?
Why do you need to convert a boolean to a number type? You only need to convert `oldYear` value.